src/libcollections/string.rs

```diff
@@ -443,15 +499,16 @@ impl String {
     /// Converts a vector of bytes to a `String` without checking that the
     /// string contains valid UTF-8.
     ///
-    /// See the safe version, [`from_utf8()`][fromutf8], for more.
+    /// See the safe version, [`from_utf8()`], for more details.
     ///
-    /// [fromutf8]: struct.String.html#method.from_utf8
+    /// [`from_utf8()`]: struct.String.html#method.from_utf8
     ///
     /// # Safety
     ///
-    /// This function is unsafe because it does not check that the bytes passed to
-    /// it are valid UTF-8. If this constraint is violated, undefined behavior
-    /// results, as the rest of Rust assumes that `String`s are valid UTF-8.
+    /// This function is unsafe because it does not check that the bytes passed
+    /// to it are valid UTF-8. If this constraint is violated, undefined
+    /// behavior results, as the rest of the standard library assumes that
```

Comment:
This may want to be reworded slightly from "undefined behavior results" to something that indicates that str's static assertions may be broken (that it contains valid utf-8) and that this could cause memory unsafety issues in downstream consumers. (although I do realize this was just reflowed a bit)